src/Altinn.App.Core/Helpers/PlatformHttpResponseSnapshotException.cs

```diff
@@ -0,0 +1,204 @@
+    /// </summary>
+    public string? ReasonPhrase { get; }
+
+    /// <summary>
+    /// Gets the HTTP version used by the response (e.g. "1.1", "2.0").
+    /// </summary>
+    public string HttpVersion { get; }
+
+    /// <summary>
+    /// Gets a flattened string representation of all response, content, and trailing headers.
+    /// </summary>
+    public string Headers { get; }
+
+    /// <summary>
+    /// Gets the response body content as a string (possibly truncated).
+    /// </summary>
+    public string Content { get; }
+
+    /// <summary>
+    /// Gets a value indicating whether the content was truncated due to the configured maximum length.
+    /// </summary>
+    public bool ContentTruncated { get; }
+
+    /// <summary>
+    /// Creates a new <see cref="PlatformHttpResponseSnapshotException"/> by snapshotting
+    /// the provided <see cref="HttpResponseMessage"/> into immutable string values,
+    /// constructing a sanitized clone for the base class, and then disposing the original response.
+    /// </summary>
+    /// <param name="response">The HTTP response to snapshot and dispose.</param>
+    /// <param name="cancellationToken">A cancellation token to cancel reading the content.</param>
+    /// <returns>The constructed <see cref="PlatformHttpResponseSnapshotException"/>.</returns>
+    public static async Task<PlatformHttpResponseSnapshotException> CreateAndDisposeHttpResponse(
+        HttpResponseMessage response,
+        CancellationToken cancellationToken = default
+    )
+    {
+        ArgumentNullException.ThrowIfNull(response);
+
+        try
+        {
+            // Snapshot content first (handle null)
+            string content = response.Content is null
+                ? string.Empty
+                : await response.Content.ReadAsStringAsync(cancellationToken);
+
+            bool truncated = content.Length > MaxCapturedContentLength;
+            if (truncated)
+            {
+                content = content[..MaxCapturedContentLength];
+            }
+
+            string headers = FlattenHeaders(response.Headers, response.Content?.Headers, response.TrailingHeaders);
+            string message = BuildMessage((int)response.StatusCode, response.ReasonPhrase, content, truncated);
+
+            // Build a sanitized, non-streaming HttpResponseMessage for the base class
+            var safeResponse = new HttpResponseMessage(response.StatusCode)
+            {
+                ReasonPhrase = response.ReasonPhrase,
+                Version = response.Version,
+            };
+
+            // Copy normal headers
+            foreach (KeyValuePair<string, IEnumerable<string>> h in response.Headers)
+            {
+                safeResponse.Headers.TryAddWithoutValidation(h.Key, h.Value);
+            }
+
+            // Attach a diagnostic snapshot body for legacy consumers (text only, truncated)
+            string mediaType = response.Content?.Headers?.ContentType?.MediaType ?? "text/plain";
```

Comment:
Probably shouldn't default to anything here if the point is to build a completely equal response message